pkg/vulnsrc/azure/oval/oval.go

```diff
@@ -5,46 +5,50 @@ import (
 	"io"
 	"path/filepath"
 
-	"golang.org/x/xerrors"
+	"github.com/samber/oops"
 
 	"github.com/aquasecurity/trivy-db/pkg/utils"
 )
 
 func ParseDefinitions(dir string) ([]Definition, error) {
 	dir = filepath.Join(dir, "definitions")
+	eb := oops.In("azure").Tags("oval").With("dir", dir)
+
 	if exists, _ := utils.Exists(dir); !exists {
-		return nil, xerrors.Errorf("no definitions dir")
+		return nil, eb.Errorf("no definitions dir")
 	}
 
 	var defs []Definition
 
 	err := utils.FileWalk(dir, func(r io.Reader, path string) error {
 		var def Definition
 		if err := json.NewDecoder(r).Decode(&def); err != nil {
-			return xerrors.Errorf("failed to decode %s: %w", path, err)
+			return eb.With("file_path", path).Wrapf(err, "failed to decode")
 		}
 		defs = append(defs, def)
 		return nil
 	})
 	if err != nil {
-		return nil, xerrors.Errorf("Azure Linux OVAL walk error: %w", err)
+		return nil, eb.Wrapf(err, "walk error")
 	}
 
 	return defs, nil
 }
 
 func ParseTests(dir string) (Tests, error) {
+	eb := oops.In("azure").Tags("oval").With("dir", dir)
 	var tests Tests
 	if err := utils.UnmarshalJSONFile(&tests, filepath.Join(dir, "tests", "tests.json")); err != nil {
-		return tests, xerrors.Errorf("failed to unmarshal tests: %w", err)
+		return tests, eb.Wrapf(err, "json unmarshal error")
```

Comment:
OopsErrorBuilder is still auxiliary code
I would prefer not to "draw unnecessary attention" to this variable (not to create `eb` variable) if the error is used once
```suggestion
		return tests, oops.In("azure").Tags("oval").With("dir", dir).Wrapf(err, "json unmarshal error")
```
same for `ParseObjects` and `ParseStates`

Reply:
For public functions, I always add domains at the top of the function, otherwise, we can easily forget to add that when we add more lines.

I documented it in the PR description. Please let me know if you have a better idea for guidelines.
https://github.com/aquasecurity/trivy-db/pull/488

Reply:
Okay, it make sense, let's use your way.